### PR TITLE
Add validation for origin country diagnosis key

### PR DESCRIPTION
**Pull request by ioangut**
## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] Make sure `mvn install` runs for the whole project and, if you touched any code in the respective service, submission and distribution service can be run with `spring-boot:run`

## Description

Enhance validation for origin country of diagnosis key.

### CODE REVIEW

**Review by EugenM-SAP — DISMISSED** [review]

File: common/persistence/src/test/java/app/coronawarn/server/common/persistence/domain/DiagnosisKeyBuilderTest.java [hunk 1]
```diff
@@ -175,14 +175,15 @@ void failsForInvalidRollingStartIntervalNumber() {
     ).isInstanceOf(InvalidDiagnosisKeyException.class);
   }
 
-  @Test
-  void failsForInvalidOriginCountry() {
+  @ParameterizedTest
+  @ValueSource(strings = {"DER", "xx", "De", "dE", "DE,FRE"})
```
Comment (EvgeniiSkrebtcov) [inline common/persistence/src/test/java/app/coronawarn/server/common/persistence/domain/DiagnosisKeyBuilderTest.java:179]: @ioangut , shall we also test for an empty string?

**Review by EvgeniiSkrebtcov — CHANGES_REQUESTED** [review]

**Review by hilmarf — DISMISSED** [review]

File: common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/validation/CountryValidator.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package app.coronawarn.server.common.persistence.domain.validation;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+
+public class CountryValidator {
```
Comment (mibrasap) [inline common/persistence/src/main/java/app/coronawarn/server/common/persistence/domain/validation/CountryValidator.java:7]: Please fix the code smell and add a private constructor to this utility class

**Review by mibrasap — APPROVED** [review]

**Review by mibrasap — APPROVED** [review]

**Review by hilmarf — APPROVED** [review]

**Review by EvgeniiSkrebtcov — APPROVED** [review]
